### DISCUSSION TITLE
[Messenger] Multiple consumers and long message handling.

### DISCUSSION THREAD

**Discussion by Braxilior** [symfony/symfony#43890] [Q&A]
Hello,

I use symfony/messenger with the redis transport and two consumers.
Recently I added a new kind of message which takes several hours to be handled.
One hour after my first worker started processing, the message was dispatched again and delivered to my second worker. Both my workers were handling the same message at the same time.
From what I understood from the documentation, It happened because my first worker was considered dead because it exceeded the `redeliver_timeout`.

I would like to know what is the general idea for this use case (multiple consumers and different scales of duration).

Thanks,

**Comment by buffcode** [comment]
The [messenger docs](https://symfony.com/doc/current/messenger.html) state:
> Set `redeliver_timeout` to a greater value than your slowest message duration. Otherwise, some messages will start a second time while the first one is still being handled.

So increase the value accordingly or optimize / split the workload so that it is below.

You could also setup a second transport with different options and publish your heavy workloads there.
